launch_testing/launch_testing/test_runner.py

```diff
@@ -59,7 +59,6 @@ def __init__(self,
         self._test_tr = threading.Thread(
             target=self._run_test,
             name='test_runner_thread',
-            daemon=True
```

Comment:
@pbaughman do you know why this was a daemon thread?

As a daemon thread I sporadically get crashes on exit because this daemon thread tries to write to stdout which is already closed by the main thread (the main thread exiting will close the global file handles but not join daemon threads). See:

https://stackoverflow.com/questions/45267439/fatal-python-error-and-bufferedwriter

Also, see:

https://docs.python.org/3.7/library/threading.html#thread-objects (part about daemonic threads)

It seems to work ok, but I wanted to know if there was a reason for it before.

Reply:
Example of the error I was getting before I changed this:

```
----------------------------------------------------------------------
Ran 1 test in 0.050s

Fatal Python error: could not acquire lock for <_io.BufferedWriter name='<stderr>'> at interpreter shutdown, possibly due to daemon threads

Thread 0x00007fe5f2ddd700 (most recent call first):
  File "/usr/lib/python3.6/unittest/runner.py", line 220 in run
  File "/home/william/ros2_ws/build/launch_testing/launch_testing/test_runner.py", line 180 in _run_test
  File "/usr/lib/python3.6/threading.py", line 864 in run
  File "/usr/lib/python3.6/threading.py", line 916 in _bootstrap_inner
  File "/usr/lib/python3.6/threading.py", line 884 in _bootstrap

Current thread 0x00007fe5f7392740 (most recent call first):
[1]    21192 abort (core dumped)  python3 -m launch_testing.launch_test  --verbose
```

Reply:
@wjwwood It's a daemon thread because I didn't want it to accidentally prevent the program from exiting if you ctrl+c and the worker thread got hung and I had had problems with this in a past life, but that was for a program where it didn't matter if there was a nasty stack trace when shutting down.  It doesn't necessarily need to be.